opencensus/stats/exporters/stackdriver_exporter.py

```diff
@@ -168,40 +169,40 @@ def upload_stats(self, view_data):
             and create time series for each value
         """
         view_data_set = utils.uniq(view_data)
-        requests = self.make_request(view_data_set, MAX_TIME_SERIES_PER_UPLOAD)
-        for request in requests:
-            self.client.create_time_series(request[CONS_NAME],
-                                           request[CONS_TIME_SERIES])
-
-    def make_request(self, view_data, limit):
+        time_series_batches = self.create_batched_time_series(
+            view_data_set, MAX_TIME_SERIES_PER_UPLOAD)
+        for time_series_batch in time_series_batches:
+            self.client.create_time_series(
+                self.client.project_path(self.options.project_id),
+                time_series_batch)
+
+    def create_batched_time_series(self, view_data, batch_size):
         """ Create the data structure that will be
             sent to Stackdriver Monitoring
         """
-        requests = []
-        time_series = []
-
-        resource = self.options.resource
-        metric_prefix = self.options.metric_prefix
-        for v_data in view_data:
-            series = self.create_time_series_list(v_data, resource,
-                                                  metric_prefix)
-            time_series.extend(series)
-
-            project_id = self.options.project_id
-            request = {}
-            request[CONS_NAME] = self.client.project_path(project_id)
-            request[CONS_TIME_SERIES] = time_series
-            requests.append(request)
-
-            if len(time_series) == int(limit):
-                time_series = []
-        return requests
+        full_time_series_list = itertools.chain.from_iterable(
+            self.create_time_series_list(
+                v_data, self.options.resource, self.options.metric_prefix)
+            for v_data in view_data)
+        time_series_batches = []
+        while True:
+            batch = []
+            for _ in range(batch_size):
+                time_series = next(full_time_series_list, None)
+                if time_series is not None:
+                    batch.append(time_series)
+            if batch:
+                time_series_batches.append(batch)
+            if len(batch) < batch_size:
+                break
```

Comment:
This batching logic comes up often enough to justify adding a function for it. I moved it to `utils.window` in ad81279, let me know what you think.